From 3b913e52450789d34155b17db3a6acca84cdbd86 Mon Sep 17 00:00:00 2001
From: Soumya Roy <souroy@nvidia.com>
Date: Thu, 1 May 2025 06:12:40 +0000
Subject: [PATCH 50/56] bgpd: add total path count for bgp net in json output

Currently only vty output shows total path count for a
BGP net. This fix add that information in josn output too.

Signed-off-by: Soumya Roy <souroy@nvidia.com>
---
 bgpd/bgp_route.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
index 4c3cd42f30..fffab7d682 100644
--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -13112,6 +13112,14 @@ void route_vty_out_detail_header(struct vty *vty, struct bgp *bgp,
 			}
 			vty_out(vty, "\n");
 		}
+
+		if (json) {
+			if (incremental_print) {
+				vty_out(vty, "\"pathCount\": %d", count);
+				vty_out(vty, ",");
+			} else
+				json_object_int_add(json, "pathCount", count);
+		}
 	}
 }
 
-- 
2.39.5

